### PR TITLE
add process killer plugin

### PR DESCRIPTION
**Pull request by AmrBinBashir**
Note: The translations may not be the best.

### CODE REVIEW

File: src/common/config/process-killer-options.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { isWindows } from "../helpers/operating-system-helpers";
```
Comment (oliverschwendener) [inline src/common/config/process-killer-options.ts:1]: This function is not available anymore. Please use `getCurrentOperatingSystem()` instead and check against the `OperatingSystem` enum.

File: src/common/config/process-killer-options.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+const defaultMacOsProcessKillerOptions: ProcessKillerOptions = {
+    isEnabled: true,
+    prefix: "kill?",
+};
+
+const defaultWindowsProcessKillerOptions: ProcessKillerOptions = {
+    isEnabled: true,
+    prefix: "kill?",
+};
```
Comment (oliverschwendener) [inline src/common/config/process-killer-options.ts:17]: These options are identical, so why separating them?

File: src/common/config/process-killer-options.ts (hunk 1 above)
Comment (AmrBinBashir) [reply src/common/config/process-killer-options.ts]: Okay.

**Comment by oliverschwendener** [comment]
@AmrBinBashir ?

File: src/common/config/process-killer-options.ts (hunk 2 above)
Comment (AmrBinBashir) [reply src/common/config/process-killer-options.ts]: Idk, I just copied them from another plugin plus I wanted to support both os but I can't do that not now at least,  I will change them.

**Comment by AmrBinBashir** [comment]
I closed it by accident.

**Comment by AmrBinBashir** [comment]
@oliverschwendener I closed it by accident.

File: src/common/translation/german-translation-set.ts [hunk 3]
```diff
@@ -345,4 +345,9 @@ export const germanTranslationSet: TranslationSet = {
 
     controlPanel: "Systemsteuerung",
     controlPanelSettingsDescription: "Dieses Plugin erlaubt es dir, Elemente der Systemsteuerung schnell zu finden.",
+
+    processKiller: "Prozesskiller",
+    processKillerPrefix: "Präfix",
+    processKillerSettingsDescription: "Dieses Plugin ermöglicht es Ihnen, die laufenden Prozesse zu töten.",
```
Comment (stefnotch) [inline src/common/translation/german-translation-set.ts:351]: I'd recommend changing it to
`Dieses Plugin ermöglicht es dir, laufende Prozesse zu töten.`

Comment (AmrBinBashir) [reply src/common/translation/german-translation-set.ts]: Thanks, I changed it.

**Comment by AmrBinBashir** [comment]
Until I can have access to a mac or If someone willing to help with doing the mac version I am gonna close this.

**Pull request closed without merging**